server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java

```diff
@@ -471,16 +471,48 @@ public CompletableFuture<Revision> updatePerRolePermissions(Author author,
                         "Can't give a permission to guest for internal repository: " + repoName);
             }
         }
-
-        final JsonPointer path = JsonPointer.compile("/repos" + encodeSegment(repoName) +
-                                                     "/perRolePermissions");
-        final Change<JsonNode> change =
-                Change.ofJsonPatch(METADATA_JSON,
-                                   new ReplaceOperation(path, Jackson.valueToTree(perRolePermissions))
-                                           .toJsonNode());
         final String commitSummary =
                 "Update the role permission of the '" + repoName + "' in the project '" + projectName + '\'';
-        return metadataRepo.push(projectName, Project.REPO_DOGMA, author, commitSummary, change);
+        final ContentTransformer<JsonNode> transformer = new ContentTransformer<>(
+                METADATA_JSON, EntryType.JSON, node -> {
```

Comment:
Optional) I could see some code similar to this in other places. How about creating `RepositoryMetadataTransformer` and duplicating them?
```java
private static class RepositoryMetadataTransformer extends ContentTransformer {
    
   RepositoryMetadataTransformer(Function<RepositoryMetadata, RepositoryMetadata>) {
       ...
   }
}
```

Reply:
That's a good suggestion. 👍 
`RepositoryMetadataTransformer`, `ProjectMetadataTransformer` and `TokensTransformer ` are added. 😉